dump_offload_mgr.cpp

```diff
@@ -7,38 +7,117 @@
 namespace openpower::dump
 {
 DumpOffloadManager::DumpOffloadManager(sdbusplus::bus::bus& bus) : _bus(bus)
+{
+    // Changing a system from hmc-managed to non-hmc manged is a disruptive
+    // process (Power off the system, do some clean ups and IPL).
+    // Changing a system from non-hmc managed to hmc-manged can be done at
+    // runtime.
+    // Not creating offloader objects if system is HMC managed
+    if (isSystemHMCManaged(_bus))
+    {
+        log<level::INFO>("DumpOffloadManager HMC managed system");
+        return;
+    }
+
+    // Perform offload only if host is in running state, else add watch on
+    // BootProgress property and offload when it moves to running state.
+    //
+    // At present we do not have a system-d target which indicates host is
+    // in running target with which we could have added wait-for condition
+    // for this service to start.
+    if (isHostRunning(bus) == false)
+    {
+        log<level::ERR>("Host is not running do not offload dump, add watch"
+                        " on host state");
+        _hostStatePropWatch = std::make_unique<sdbusplus::bus::match_t>(
+            _bus,
+            sdbusplus::bus::match::rules::propertiesChanged(
+                "/xyz/openbmc_project/state/host0",
+                "xyz.openbmc_project.State.Host"),
+            [this](auto& msg) { this->propertiesChanged(msg); });
+    }
+    else
+    {
+        addOffloadHandlers();
+    }
+}
+
+void DumpOffloadManager::propertiesChanged(sdbusplus::message::message& msg)
+{
+    std::string intf;
+    DBusPropertiesMap propMap;
+    msg.read(intf, propMap);
+    log<level::DEBUG>(
+        fmt::format("propertiesChanged interface ({}) ", intf).c_str());
+    for (auto prop : propMap)
+    {
+        if (prop.first == "BootProgress")
+        {
+            auto progress = std::get_if<std::string>(&prop.second);
+            if (progress != nullptr)
+            {
+                ProgressStages bootProgress =
+                    sdbusplus::xyz::openbmc_project::State::Boot::server::
+                        Progress::convertProgressStagesFromString(*progress);
+                if ((bootProgress == ProgressStages::SystemInitComplete) ||
+                    (bootProgress == ProgressStages::OSStart) ||
+                    (bootProgress == ProgressStages::OSRunning))
+                {
+                    // we are checking for multiple values as part of
```

Comment:
now the logic is changed every time system goes to running it will try to check if any dumps are there and offload them. if no dumps nothing will be offloaded.